react/README.md

```diff
@@ -0,0 +1,477 @@
+
+## Class vs `React.createClass` vs stateless
+
+  - Prefer `class extends React.Component` over `React.createClass`. In practice, most components should extend either `AppContextComponent` or `PureRenderComponent` (`Component` for React Native).
+
+    ```jsx
+    // bad
+    const Listing = React.createClass({
+      // ...
+      render() {
+        return <div>{this.state.hello}</div>;
+      }
+    });
+
+    // good
+    export default class Listing extends PureRenderComponent {
+      // ...
+      render () {
+        return <div>{this.state.hello}</div>;
+      }
+    }
+    ```
+
+## Naming
+
+  - **Extensions**: Use `.js` extension for React components.
+  - **Filename**: Use camelCase for filenames. E.g., `reservationCard.js`.
+  - **Reference Naming**: Use PascalCase for React components and camelCase for their instances. 
+
+    ```jsx
+    // bad
+    import reservationCard from './reservationCard';
+
+    // good
+    import ReservationCard from './reservationCard';
+
+    // bad
+    const ReservationItem = <ReservationCard />;
+
+    // good
+    const reservationItem = <ReservationCard />;
+    ```
+
+  - **Component Naming**: Use the filename as the component name. For example, `cardFeed.js` should have a reference name of `CardFeed`. Namespace subcomponents with the main component name followed by a hyphen.
+
+    ```jsx
+    // bad
+    import ContributionCard from './card/contribution';
+    import UserCard from './card/user';
+
+    // good
+    import ContributionCard from './card/card-contribution';
+    import UserCard from './card/card-user';
+    ```
+
+## Declaration
+
+  - Do not use `displayName` for naming components. Instead, name the component by reference.
+
+    ```jsx
+    // bad
+    export default React.createClass({
+      displayName: 'ReservationCard',
+      // stuff goes here
+    });
+
+    // good
+    export default class ReservationCard extends PureRenderComponent {
+      // stuff goes here
+    }
+    ```
+
+## Alignment
+
+  - Follow these alignment styles for JSX syntax.
+
+    ```jsx
+    // bad
+    <Foo
+      superLongParam="bar"
+      anotherSuperLongParam="baz"
+    />
+
+    // good
+    <Foo superLongParam="bar"
+      anotherSuperLongParam="baz" />
+
+    // if props fit in one line then keep it on the same line
+    <Foo bar="bar" />
+
+    // children get indented normally
+    <Foo
+      superLongParam="bar"
+      anotherSuperLongParam="baz">
+      <Quux />
+    </Foo>
+    ```
+
+## Quotes
+
+  - Always use double quotes (`"`) for JSX attributes, but single quotes for all other JS. eslint: [`jsx-quotes`](http://eslint.org/docs/rules/jsx-quotes)
+
+  > Why? JSX attributes [can't contain escaped quotes](http://eslint.org/docs/rules/jsx-quotes), so double quotes make conjunctions like `"don't"` easier to type.
+  > Regular HTML attributes also typically use double quotes instead of single, so JSX attributes mirror this convention.
```

Comment:
I get this, and I don’t hate it; but I feel we should update this section to reflect our consistent use of single quotes throughout our JS files. (Besides, if you use [proper apostrophes](http://smartquotesforsmartpeople.com/), you don’t have to worry about escaping single quotes. :trollface:)